Examples/Io/Root/src/RootTrackSummaryWriter.cpp

```diff
@@ -131,18 +107,41 @@ RootTrackSummaryWriter::RootTrackSummaryWriter(
   m_outputTree->Branch("err_eTHETA_fit", &m_err_eTHETA_fit);
   m_outputTree->Branch("err_eQOP_fit", &m_err_eQOP_fit);
   m_outputTree->Branch("err_eT_fit", &m_err_eT_fit);
-  m_outputTree->Branch("res_eLOC0_fit", &m_res_eLOC0_fit);
-  m_outputTree->Branch("res_eLOC1_fit", &m_res_eLOC1_fit);
-  m_outputTree->Branch("res_ePHI_fit", &m_res_ePHI_fit);
-  m_outputTree->Branch("res_eTHETA_fit", &m_res_eTHETA_fit);
-  m_outputTree->Branch("res_eQOP_fit", &m_res_eQOP_fit);
-  m_outputTree->Branch("res_eT_fit", &m_res_eT_fit);
-  m_outputTree->Branch("pull_eLOC0_fit", &m_pull_eLOC0_fit);
-  m_outputTree->Branch("pull_eLOC1_fit", &m_pull_eLOC1_fit);
-  m_outputTree->Branch("pull_ePHI_fit", &m_pull_ePHI_fit);
-  m_outputTree->Branch("pull_eTHETA_fit", &m_pull_eTHETA_fit);
-  m_outputTree->Branch("pull_eQOP_fit", &m_pull_eQOP_fit);
-  m_outputTree->Branch("pull_eT_fit", &m_pull_eT_fit);
+
+  if (m_inputParticles.isInitialized() &&
+      m_inputTrackParticleMatching.isInitialized()) {
```

Comment:
I am always a bit unsure if we should make columns optional. This can make reading a file complicated but I am also not sure if reading really is a use case

Reply:
what do you mean by "reading" here?

Reply:
I think "when is what actually being read, needed, non-empty".

Reply:
I think figuring out which parts are set in which scenarios when reading the source code. The more optional parts you have, the trickier it gets.